### DISCUSSION TITLE
Hiding/showing block palette

### DISCUSSION THREAD

**Discussion by sfederici** [ScratchAddons/ScratchAddons#1156] [Ideas] [answered]
when I teach, tt would be very useful to me (but -I guess- it would be also useful to programmers when they have large/many scripts) to be able to hide/show the block palette. This could be done by pressing the block category button of the category shown in the palette (and the block palette could be instead made visible again when you click whatever category block)

**Comment by cobaltt7** [comment]
I thought about this too! Theres another extention that does something similar to that (I think it's called ***Better 3.0***)

Paul Reid

________________________________

**Reply by sfederici** [reply]
Cool! But I can't find it in my Scratch AddOns settings.

**Reply by WorldLanguages** [reply]
@sfederici Better 3.0 is another *extension*, it's not within Scratch Addons. https://chrome.google.com/webstore/detail/better30/gjoimmimmkpppgkhobhceblafkienddi It can be a little buggy but I think it includes the feature you mention

**Comment by sfederici** [comment]
Oh, I see. I'd rather stick to Scratch AddOns. The developer will forgive me, but I don't like to add too much stuff to my PC that I don't know if I can fully trust.

**Reply by WorldLanguages** [reply]
@sfederici Don't worry about it, the creator of Better3.0 is also a member of Scratch Addons, he designed the settings page (Ucrash), fully trusted

**Comment by TheColaber** [comment, marked as answer]
Hey, @sfederici I actually just made a PR for this suggestion to be implemented... you can see the pr here: #1319. This will most likely be an addition to version 1.8.0 of Scratch Addons

**Reply by sfederici** [reply]
Excellent! I hope versione 1.8 will be out soon!

**Comment by TheColaber** [comment]
I wanted to make sure I interpreted your post correctly... did you mean something like this?

https://user-images.githubusercontent.com/72760579/104682551-ede84b00-56c2-11eb-90ad-c655123ede4a.mp4

This action happens when the setting called `SHOW FLYOUT WHEN CATAGORY CLICKED` is enabled... otherwise, this is the action:

https://user-images.githubusercontent.com/72760579/104682737-6222ee80-56c3-11eb-8404-1937018a8ec8.mp4

**Reply by sfederici** [reply]
My vote goes to the first behavior, as I teach to newbies so it is better to have the blocks visible at start. But for experienced Scratchers the second one can be good too (even if the continuous opening and closing it is not ideal in my view)

**Reply by TheColaber** [reply]
Great! I'm glad to see I got it right!

**Reply by sfederici** [reply]
I have seen the new addon and I'm very happy of the possibility to hide the block palette. Thanks a lot! However my request was a little different, as the present implementation doesn't allow the user to jump directly from one category of blocks to another by clicking the category selectors. My request was indeed hiding the block palette "by pressing the block category button _of the category shown in the palette_ ", not of every category. In this way you can still use the buttons to jump from one category to another without having to click the block palette twice.

**Reply by WorldLanguages** [reply]
@sfederici Good report, that's how it was supposed to work but we missed it. Should be fixed with #1419

**Reply by TheColaber** [reply]
Yeah sorry about that. Somehow I had it working before, and then somewhere in the commits it changed... anyway so sorry, should be fixed for the next update.

**Reply by sfederici** [reply]
Excellent, thanks

**Reply by sfederici** [reply]
Now it works exactly as I expected it to work! Thanks a lot! Excellent job.
